### PR TITLE
Train Mask-RCNN

### PR DESCRIPTION
**Pull request by g1y5x3**
So far I am able to setup loading training data as well as iterate the `(image, target)` pair for running forward and backward.
`MODEL=maskrcnn python3 examples/mlperf/model_train.py` 
`(<Tensor <LB (3, 800, 1066) dtypes.float op=LoadOps.FROM st=ShapeTracker(shape=(3, 800, 1066), views=[View((3, 800, 1066), (852800, 1066, 1), 0, None)])> on GPU with grad None>, BoxList(num_boxes=50, image_width=1066, image_height=800, mode=xyxy), slice(0, 10, None))`

However, I think I need to refactor the code quite a bit. `models/mask_rcnn.py` is really messy and needs a lot of clean up. Ideally organize into different files according to their functionalities like what [mlcommons](https://github.com/mlcommons/training/tree/master/object_detection/pytorch/maskrcnn_benchmark/) did.

Additionally, a lot of the indexing operations have to be done with converting to numpy first and then convert back to Tensor, which was mentioned #967 and I think that's essentially why @kunwar31 implemented a bunch of additional functions just to index Tensors with different conditions. 

Finally, this is the current estimated inference time with a single NVIDIA 3090, it just felt wrong...
`1/5000 [02:10<181:21:58, 130.61s/it`

- [x] load and index through training `(image, target)` pair
- [ ] modularize `models/mask_rcnn.py` for better readability as well as remove some of the dependency on torch backend (mostly transforms) 
- [ ] find a solution for doing advanced indexing on Tensor directly
- [ ] add loss function into RPN, ROIHead
- [ ] perform training experiments

### CODE REVIEW

**Comment by wozeparrot** [comment]
Need to be careful with `.numpy()` usage cuz grads won't flow through it, `.numpy()` should be fine for preprocessing and postprocessing stuff but it shouldn't be used in the actual model.

**Comment by geohot** [comment]
Bounty locked, good luck!

File: models/mask_rcnn.py [hunk 1]
```diff
@@ -254,31 +257,389 @@ def transpose(self, method):
 
   def clip_to_image(self, remove_empty=True):
     TO_REMOVE = 1
-    bb1 = self.bbox.clip(min_=0, max_=self.size[0] - TO_REMOVE)[:, 0]
-    bb2 = self.bbox.clip(min_=0, max_=self.size[1] - TO_REMOVE)[:, 1]
-    bb3 = self.bbox.clip(min_=0, max_=self.size[0] - TO_REMOVE)[:, 2]
-    bb4 = self.bbox.clip(min_=0, max_=self.size[1] - TO_REMOVE)[:, 3]
+    # TODO find solutions for 
+    # 'Tensor' object does not support item assignment
+    # 'Tensor' does not support & (and) operation
```
Comment (kunwar31) [inline models/mask_rcnn.py:262]: & operation is just elementwise multiply for 0/1 Tensors

**Comment by g1y5x3** [comment]
lol made a mistake
